### PR TITLE
Use tc-lib-validate in place of schema-validator-publisher

### PR DESCRIPTION
**Pull request by imbstack**
This is an api breaking change, but introduces some nice errors in place of the opaque ones that the old library had.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "taskcluster-client": "0.23.4",
     "taskcluster-lib-scopes": "^0.8.8",
     "taskcluster-lib-stats": "^0.8.7",
-    "taskcluster-lib-testing": "^0.10.1",
+    "taskcluster-lib-validate": "^0.4.2",
```
Comment (jonasfj) [inline package.json]: isn't this only a devDepenednecy?

File: src/api.js [hunk 2]
```diff
@@ -111,15 +112,15 @@ var schema = function(validator, options) {
...
+      var error = validate(req.body, options.input);
+      if (error) {
         debug("Request payload for %s didn't follow schema %s",
               req.url, options.input);
         return res.reportError(
           'InputValidationError',
           "Request payload must follow the schema: {{schema}}\n" +
-          "Errors: {{errors}}", {
-          errors,
+          "Error: {{error}}", {
```
Comment (jonasfj) [inline src/api.js]: I suspect we don't need the `Error: ` part... is the `error` variable a full error message with everything needed?

So maybe `res.reportError('InputValidationError', error, {schema: options.input, payload: req.body});` is fine...

File: src/api.js [hunk 3]
```diff
@@ -131,14 +132,14 @@ var schema = function(validator, options) {
...
       // defined, then we have to validate against it...
       if(options.output !== undefined && !options.skipOutputValidation) {
-        var errors = validator.check(json, options.output);
-        if (errors) {
+        var error = validate(json, options.output);
+        if (error) {
           debug("Reply for %s didn't match schema: %s got errors: %j from output: %j",
-                req.url, options.output, errors, json);
+                req.url, options.output, error, json);
           let err = new Error('Output schema validation of ' + options.output);
```
Comment (jonasfj) [inline src/api.js]: I propose we do: ` new Error('Output schema validation error: ' + error)` as the message is pretty featureful right?

File: package.json (hunk 1 above)
Comment (imbstack) [inline package.json]: I'm a bit confused. I didn't think this was devDependency only. schema-validator-publisher wasn't. Was that a mistake?

Comment (imbstack) [inline package.json]: It's also imported in `src/api.js` so that seems like a real dependency to me.

Comment (jonasfj) [inline package.json]: nope...
But we get validate passed in now... So we don't need it as a dependency, right?

Comment (jonasfj) [inline package.json]: We also had it passed in before... the only place we used package was in reference() where you use ajv now...

**Comment by jonasfj** [comment]
`r+`, but please make sure the error messages are as good as they can get see my comments..

File: package.json (hunk 1 above)
Comment (imbstack) [inline package.json]: Ah, yes. Duh.

Comment (jonasfj) [inline package.json]: but not used in src/api.js anymore..

**Pull request merged**